subconscious/model.py

```diff
@@ -314,6 +326,25 @@ def get_index_key(cls, column_name):
                 first_iteration = False
             else:
                 result_set = result_set.intersection(temp_set)
+            if order_by:
+                pairs = []
+
+                def _pair_with_zero_score(x):
+                    return [0, x]
+
+                [pairs.extend(_pair_with_zero_score(x)) for x in result_set]
+                if pairs:
+                    filtered_res_key = 'filtered_result-{}'.format(uuid.uuid1())
+                    await db.zadd(filtered_res_key, pairs[0], pairs[1], *pairs[2:])
+                    sorted_result = await db.sort(
+                        filtered_res_key,
+                        by='{}:*->{}'.format(cls.__name__, order_by),
+                        alpha=True,
+                        asc=direction
+                    )
+                    # Delete the temp store
+                    await db.delete(filtered_res_key)
+                    return sorted_result
```

Comment:
With this way redis will sort the ids for us without needing to retrieve all the model objects , also if we retrieve all objects for our sorting that will defeat the purpose of async generators for our queries.